src/submission/validation.ts

```diff
@@ -140,6 +146,68 @@ export const usingInvalidProgramId = (
   }
   return [];
 };
+
+export const checkUniqueRecords = (
+  clinicalType: ClinicalEntityType,
+  newRecords: DeepReadonly<DataRecord[] | CreateRegistrationRecord[]>,
+  useAllRecordValues?: boolean, // use all record properties so it behaves like duplicate check
```

Comment:
is this used in any case, I see you picked sample_submitter_id as unique identifier for registration ?

Reply:
nvm I see that above